app/jobs/google/search_keyword_job.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Google
+  class SearchKeywordJob < ApplicationJob
+    queue_as :default
+
+    def perform(keyword)
+      search_result = Google::SearchService.new(keyword[:name]).call
+      raise Google::Errors::SearchKeywordError unless search_result
+
+      update_keyword(keyword, search_result)
+    end
+
+    private
+
+    def update_keyword(keyword, search_result)
+      source = Source.find_or_create_by({ name: 'Google' })
+      keyword.update(
+        top_ads_count: search_result[:ads_top_count],
+        total_ads_count: search_result[:ads_page_count],
+        ads_links: search_result[:ads_top_urls],
+        result_count: search_result[:result_count],
+        result_links: search_result[:result_urls],
+        total_link_count: search_result[:total_link_count],
+        html: search_result[:html],
+        source: source
+      )
+    end
```

Comment:
```suggestion
    def update_keyword(keyword, search_result)
      keyword.update search_attributes
    end
    
    def search_attributes
      {
        top_ads_count: search_result[:ads_top_count],
        total_ads_count: search_result[:ads_page_count],
        ads_links: search_result[:ads_top_urls],
        result_count: search_result[:result_count],
        result_links: search_result[:result_urls],
        total_link_count: search_result[:total_link_count],
        html: search_result[:html],
        source: Source.find_or_create_by({ name: 'Google' })
      }
    end
```

The `lines` limit is not a big problem, as long as the long-methods are a single hash/array/... i.e. they have a very specific responsibility.

In this case, I would recommend extracting the hash build into a dedicated method (even if this dedicated method would still be more than 10 lines, it would be OK).


## But I have another question: 

> Why do we have different keys in `keyword` and `search_result`?

For example, in the line: `top_ads_count: search_result[:ads_top_count],`, why `top_ads_count` is not the same as `ads_top_count`? 🤔 

If we adopt the same keys (*which I do recommend!*), we could write the code in a more elegant way: 

```
    def update_keyword(keyword, search_result)
      source = Source.find_or_create_by({ name: 'Google' })

      keyword.update(search_result.merge(source: source))
    end
```

In case `search_result` contains some unwanted keys, you can either `except` them, or `slice` all the desired keys using a constant that lists all the keys we want to use. But I think `except/slice` is not needed in that case 🤔

Reply:
Updated this to allow easier call on the job.

Reply:
Clean ✨